outwatch/src/main/scala/outwatch/dom/Handler.scala

```diff
@@ -1,34 +1,28 @@
 package outwatch
+package dom
 
-import monix.execution.{Ack, Scheduler}
-import monix.reactive.subjects.PublishSubject
+import cats.effect.Sync
+import cats.implicits._
+import monix.execution.{Ack, Cancelable, Scheduler}
+import monix.reactive.observers.Subscriber
+import monix.reactive.subjects.{BehaviorSubject, ReplaySubject}
 import monix.reactive.{Observable, Observer}
 
 import scala.concurrent.Future
 
-trait MonixOps {
-  type ProHandler[-I, +O] = Observable[O] with Observer[I]
-  type Handler[T] = ProHandler[T,T]
+trait HandlerOps[F[_]] {
```

Comment:
My problem is that I am not sure which effect type we should choose by default for `Handler.create` (this is about #321). If we want to handle `IO` async in the future, we will have one extra patch wherever we use `IO`. For `SyncIO`, we can do it synchronously, which is much better for constructing the `VNode` for snabbdom patching, because the initial patch contains everything after the effect is run. Whereas async effects can only be patched in afterwards.

What I am asking is whether it makes sense to have `Handler.create[F, T]` instead of `trait Handler[F[_]] { def create[T] }`? I think, it can be useful for mixing different effect types in the same app.

Reply:
I think it makes sense, the only problem I have with moving the `F` into the method is that it breaks type inference. Since `Handler.create` doesn't accept an argument with an `F`, it means you'll always have to specify the full return type as a type argument or between `:` and `=`.  This might even be desirable in this case, not sure.

Reply:
Another big point here is that it doesn't really make sense for `Handler.create` or `ProHandler.create` to return inside of `F`. All these methods are doing is `Sync[F].delay {  ... }` where nothing in `...` cares about `F`. So we could simply leave it up to the caller to create the handler inside of an F if needed.

Reply:
I pushed up a commit that looks at removing F from ProHandler and Handler, 

https://github.com/zakpatterson/outwatch/commit/54db1632b955a052c884b48b04cf7e8320a4ded7

I also refactored all the test code in this commit (painful), to show how that would look. I thought I'd get your feedback before merging that into this PR. It does seem a bit simpler to me at least, to just have the `Handler.create` etc return the Handler without wrapping in `F`.

Reply:
That is an excellent point to bring up. The advantage of enforcing an effect for `Handler.create` is that users are forced to building referentially transparent components/functions when handling state in the form of `Handler` in your app.

Take for example:

```scala
def counter = Handler.create[Int].map(handler => div(handler, onClick(handler.map(_ + 1)) --> handler)
div(counter, counter) // you get two separate counter components
```

Now, we have a method without wrapping in an effect called `Handler.unsafe` which is at least explicit in the naming. 

```scala
def counter = {
  val handler = Handler.unsafe[Int]
  div(handler, onClick(handler.map(_ + 1)) --> handler)
}
div(counter, counter) // you get two dependent counter components that share the same state
```

Apart from that, some users just might not care about these properties and like to just handlers without IO. So your argument to wrap it yourself is actually quite good, it kind of becomes more flexible. But the question is whether it is explicit enough.

@LukaJCB What do you think?

Reply:
Handler.unsafe is just an alias for this: https://github.com/monix/monix/blob/v3.0.0-RC3/monix-reactive/shared/src/main/scala/monix/reactive/subjects/ReplaySubject.scala#L214. Why should that even be considered unsafe?

Reply:
As, I tried to describe in my previous comment, defining the method `Handler.create` without IO would not be referential transparent. It returns a stateful interface (`Observer`).

If I would define create like:

```scala
def create[T] = PublishSubject[T]
div(counter(create[Int]), counter(create[Int]))
```

then it is not equivalent  to:

```scala
def create[T] = PublishSubject[T]
val handler = create[Int]
div(counter(handler), counter(handler))
```

Reply:
With IO and without, you would still have two different handlers.  I think in both cases those two examples are not equivalent.

Reply:
Without `IO`, you do not have different handler (if you use the return value twice instead of calling the function twice). With  `IO`, it is equivalent, because in both cases you get a different handler.

Reply:
That’s right, sorry. thanks.

Reply:
Whereas with iO if you wanted two counters showing the same state, you would map over IO and pass the inner value twice.

Reply:
Thanks for pushing back on this one. Another concern I had was the type inference issue, but I solved that with the partial application trick.  

So now you don't have to write `Store.create[IO, Action, Model](intiialAction, model)`. You can just write `Store.create[IO](initialAction, model)`. Same with `Handler.create`.  See  
d80a55b.